tests/test_basis.h

```diff
@@ -31,10 +31,258 @@
 #ifndef TEST_BASIS_H
 #define TEST_BASIS_H
 
-#include "core/os/main_loop.h"
+#include "core/math/random_number_generator.h"
+#include "core/os/os.h"
+#include "core/ustring.h"
+
+#include "tests/test_macros.h"
 
 namespace TestBasis {
-MainLoop *test();
+
+enum RotOrder {
+	EulerXYZ,
+	EulerXZY,
+	EulerYZX,
+	EulerYXZ,
+	EulerZXY,
+	EulerZYX
+};
+
+Vector3 deg2rad(const Vector3 &p_rotation) {
+	return p_rotation / 180.0 * Math_PI;
+}
+
+Vector3 rad2deg(const Vector3 &p_rotation) {
+	return p_rotation / Math_PI * 180.0;
+}
+
+Basis EulerToBasis(RotOrder mode, const Vector3 &p_rotation) {
+	Basis ret;
+	switch (mode) {
+		case EulerXYZ:
+			ret.set_euler_xyz(p_rotation);
+			break;
+
+		case EulerXZY:
+			ret.set_euler_xzy(p_rotation);
+			break;
+
+		case EulerYZX:
+			ret.set_euler_yzx(p_rotation);
+			break;
+
+		case EulerYXZ:
+			ret.set_euler_yxz(p_rotation);
+			break;
+
+		case EulerZXY:
+			ret.set_euler_zxy(p_rotation);
+			break;
+
+		case EulerZYX:
+			ret.set_euler_zyx(p_rotation);
+			break;
+
+		default:
+			// If you land here, Please integrate all rotation orders.
+			REQUIRE_MESSAGE(false, "This is not unreachable.");
```

Comment:
You can use `FAIL` with a message which fails an entire test suite without running any remaining assertions.
And probably would better to encode the comment directly in there:
```suggestion
			FAIL("If you land here, please integrate all rotation orders.");
```